kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java

```diff
@@ -631,6 +632,176 @@ public void testUpdateContainerWithDifferrentID() {
         assertEquals(releaseId, containerResponseEntity.getReleasedId());
         assertEquals(KieContainerStatus.STOPPED, containerResponseEntity.getStatus());
     }
+    
+    @Test
+    public void testUpdateContainerNonValidReleaseId() throws Exception {
+        // Create kie server instance connection in controller.
+        ServerTemplate serverTemplate = createServerTemplate();
+
+        // Deploy container for kie server instance.
+        ContainerSpec containerToDeploy = new ContainerSpec(CONTAINER_ID, CONTAINER_NAME, serverTemplate, releaseId, KieContainerStatus.STARTED, Collections.EMPTY_MAP);
+        mgmtControllerClient.saveContainerSpec(serverTemplate.getId(), containerToDeploy);
+
+        // Get container using kie controller.
+        ContainerSpec containerResponseEntity = mgmtControllerClient.getContainerInfo(kieServerInfo.getServerId(), CONTAINER_ID);
+        checkContainer(containerResponseEntity, KieContainerStatus.STARTED);
+        assertEquals(CONTAINER_NAME, containerResponseEntity.getContainerName());
+
+        // Check that container is deployed in kie server.
+        KieServerSynchronization.waitForKieServerSynchronization(client, 1);
+        ServiceResponse<KieContainerResource> containerInfoResponse = client.getContainerInfo(CONTAINER_ID);
+        assertEquals(ServiceResponse.ResponseType.SUCCESS, containerInfoResponse.getType());
+        KieContainerResource containerResource = containerInfoResponse.getResult();
+        assertEquals(CONTAINER_ID, containerResource.getContainerId());
+        assertEquals(KieContainerStatus.STARTED, containerResource.getStatus());
+        assertEquals(releaseId, containerResource.getReleaseId());
+
+        // Update container
+        ReleaseId nonValidReleaseId = new ReleaseId("org.kie.server.testing", "stateless-session-kjar", "2.0.0-SNAPSHOT");
+        containerToDeploy.setReleasedId(nonValidReleaseId);
+
+        // We can update container to new version, but if can't be found, then deployed container is not changed.
+        mgmtControllerClient.updateContainerSpec(serverTemplate.getId(), containerToDeploy);
+
+        // Get container using kie controller. Container has changed ReleaseId
+        containerResponseEntity = mgmtControllerClient.getContainerInfo(kieServerInfo.getServerId(), CONTAINER_ID);
+        assertEquals(CONTAINER_ID, containerResponseEntity.getId());
+        assertEquals(CONTAINER_NAME, containerResponseEntity.getContainerName());
+        assertEquals(KieContainerStatus.STARTED, containerResponseEntity.getStatus());
+        assertEquals(nonValidReleaseId, containerResponseEntity.getReleasedId());
```

Comment:
@mswiderski I have a question/idea. If container update fails on some Kie server (like here when release id is not valid) then I would expect controller to fall back to last running release id. Otherwise we may run into inconsistency. What do you think?